Slack&whatsapp-notifier.py

```diff
@@ -94,72 +85,60 @@ def execute_query(mycursor, query, t):
-        print("SLACK MESSAGE IS SENT")
-
+        logging.info("SLACK MESSAGE IS SENT")
     except slack.errors.SlackApiError as e:
-        print(f"Error sending Slack message: {e.response['error']}")
+        logging.error(f"Error sending Slack message: {e.response['error']}")
 
 # Sending WhatsApp message
 def send_whatsapp_message(message, th, tm):
-    '''pywhatkit.sendwhatmsg("+91 9310830655", message, th, tm + 1, 15, True, 2)
-    print("WhatsApp message is sent.")
-    return'''
+    #number = os.getenv('MOBILE_NUMBER')
+    # pywhatkit.sendwhatmsg(number, message, th, tm + 1, 15, True, 2)
+    # logging.info("WhatsApp message is sent.")
+    return
 
 def main():
-    while True:
-
-        load_environment() #To load envirounment variables
-        conn = get_db_connection() # Connect to the database
-        mycursor = conn.cursor()
-        data= extracting_data_from_json() # Extracting data from json file
-        for i in range (0,len(data)):
-            if (data[i]['type']=="persistant"):
-                database_name, table_name, target_coloumn, condition, target_value, time_column = if_type_persistant(i, data)
-                open_database(database_name, mycursor) # Opening of database
-                query = Persistant_building_of_query(target_coloumn, table_name, time_column)
-                t=time.localtime().tm_hour #To get current time
-                result = execute_query(mycursor, query,t)
-                print("NO. of transactions in last hour are:", result)
-                
-            elif (data[i]['type']=='threshold'):
-                database_name, table_name, threshold, time_column = if_type_threshold(i, data)
-                open_database(database_name, mycursor) # Opening of database   
-                WHERE = Threshold_WHERE_statement(target_coloumn, condition, target_value, time_column)
-                query = Threshold_building_of_query(target_coloumn, table_name, WHERE)
-                t=time.localtime().tm_hour #To get current time
-                result = execute_query(mycursor, query,t)
-                process_failed_transactions(result, threshold)
-
-        # Close the connection
-        conn.close()
-
-        # Wait for an hour before the next cycle
-        time.sleep(3600)
-
+    setup_logs()
+    load_environment()
+    conn = get_db_connection()
+    mycursor = conn.cursor()
+    data = extracting_data_from_json()
+    
+    for i in range(len(data)):
+        database_name, table_name, time_column, target_coloumn, condition, target_value = required_field_json(i, data)
+        open_database(database_name, mycursor)
+        WHERE = where_statement(target_coloumn, condition, target_value, time_column)
+        query = building_of_query(target_coloumn, table_name, WHERE)
+        t = time.localtime().tm_hour
+        result = execute_query(mycursor, query, t)
+        if data[i]['type']=='persistant' :
+            logging.info(f"No. of transactions in last hour are: {result}")

```

Comment:
where are we sending the notification for persistent query?

Reply:
if the type of query is persistent then, we are just logging the count of the queries which satisfies the condition